persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOTest.kt

```diff
@@ -437,6 +441,61 @@ class MessageDAOTest : BaseDatabaseTest() {
         assertContains(messageTypes, UnreadEventTypeEntity.MESSAGE)
     }
 
+    @Test
+    fun givenAssetMessage_WhenAssetMessages_ThenListShouldContainAssetMessage() = runTest {
+        // given
+        val domain = "domain"
+        val conversationId = QualifiedIDEntity("1", domain)
+        val assetId = "assetId"
+        val messageId = "assetMessage"
+        conversationDAO.insertConversation(
+            newConversationEntity(id = conversationId)
+        )
+
+        userDAO.upsertUser(userEntity1)
+
+        messageDAO.insertOrIgnoreMessages(
+            listOf(
+                newRegularMessageEntity(
+                    id = messageId,
+                    date = "2000-01-01T13:00:00.000Z".toInstant(),
+                    conversationId = conversationId,
+                    senderUserId = userEntity1.id,
+                    content = MessageEntityContent.Asset(
+                        1000,
+                        assetName = "test name",
+                        assetMimeType = "image/png",
+                        assetDownloadStatus = MessageEntity.DownloadStatus.SAVED_INTERNALLY,
+                        assetOtrKey = byteArrayOf(1),
+                        assetSha256Key = byteArrayOf(1),
+                        assetId = assetId,
+                        assetToken = "",
+                        assetDomain = domain,
+                        assetEncryptionAlgorithm = "",
+                        assetWidth = 111,
+                        assetHeight = 111,
+                    )
+                )
+            )
+        )
+
+        assetDao.insertAsset(
+            AssetEntity(
+                key = assetId,
+                dataSize = 123,
+                dataPath = "dataPath",
+                domain = domain,
+                downloadedDate = null
+            )
+        )
+
+        // when
+        val assetMessages = messageDAO.getMessageAssets(conversationId, 100, 0)
```

Comment:
question: can it be tested a bit more? It uses query that for instance filters out assets having dimensions equal to 0, non null id, non ephemeral, so maybe we could test if this query returns only valid assets